x-pack/legacy/plugins/rollup/public/crud_app/sections/job_create/steps/step_metrics.js

```diff
@@ -47,114 +43,300 @@ const whiteListedMetricByFieldType = {
+
+  renderMetricsSelectAllCheckboxes() {
+    const {
+      fields: { metrics },
+      onFieldsChange,
+    } = this.props;
 
-          if (!isAllowed) {
-            return;
+    let disabledCheckboxesCount = 0;
+
+    /**
+     * Look at all the metric configs and include the special "All" checkbox which adds the ability
+     * to select a all the checkboxes across columns and rows.
+     */
+    const jsxElements = [{
+      label: i18n.translate('xpack.rollupJobs.create.stepMetrics.allCheckbox', { defaultMessage: 'All' }),
+      type: 'all',
+    }].concat(metricTypesConfig).map(({ label, type: metricType, fieldTypes }, idx) => {
+      const isAllMetricTypes = metricType === 'all';
+      // For this config we are either considering all user selected metrics or a subset.
+      const applicableMetrics = isAllMetricTypes
+        ? metrics
+        : metrics
+          .filter(({ type }) => {
+            return fieldTypes[type];
+          });
+
+      let checkedCount = 0;
+      let isChecked = false;
+      let isDisabled = false;
+
+      if (isAllMetricTypes) {
+        applicableMetrics.forEach(({ types, type }) => {
+          const whiteListedSubset = Object.keys(whiteListedMetricByFieldType[type]);
+          if (whiteListedSubset.every(metricName => types.some(type => type === metricName))) {
+            ++checkedCount;
+          }
+        });
+
+        isDisabled = metrics.length === 0;
+      } else {
+        applicableMetrics.forEach(({ types }) => {
+          const metricSelected = types.some(type => type === metricType);
+          if (metricSelected) {
+            ++checkedCount;
           }
+        });
+
+        isDisabled = !metrics.some(({ type: fieldType }) =>
+          checkWhiteListedMetricByFieldType(fieldType, metricType),
+        );
+      }
+
+      // Determine if a select all checkbox is checked.
+      isChecked = checkedCount === applicableMetrics.length;
+
+      if (isDisabled) ++disabledCheckboxesCount;
+
+      return (
+        <EuiCheckbox
+          id={`${idx}-select-all-checkbox`}
+          data-test-subj={`rollupJobMetricsSelectAllCheckbox-${metricType}`}
+          disabled={isDisabled}
+          label={<span style={{ whiteSpace: 'nowrap' }}>{label}</span>}
+          checked={!isDisabled && isChecked}
+          onChange={() => {
+            if (isAllMetricTypes) {
+              const newMetrics = metricTypesConfig
+                .reduce((acc, { type }) => this.setMetrics(type, !isChecked), null);
+              onFieldsChange({ metrics: newMetrics });
+            } else {
+              onFieldsChange({ metrics: this.setMetrics(metricType, !isChecked) });
+            }
+          }}
+        />
+      );
+    });
 
-          const isSelected = types.includes(type);
+    return {
+      jsxElements,
+      allCheckboxesDisabled: jsxElements.length === disabledCheckboxesCount,
+    };
+  }
+
+  getMetricsSelectAllMenu() {
+    const {
+      jsxElements: checkboxElements,
```

Comment:
would it make sense to rename `jsxElements` to `checkboxElements` in `renderMetricsSelectAllCheckboxes()`?